### PR TITLE
Make interpreters code async by default

### PR DESCRIPTION
**Pull request by WebReflection**
This MR would like to make all interpreters run `async` code by default. The current logic is that if the `async` attribute is present and its value is explicitly `false`  then it goes back to `sync` mode, otherwise all code runs just with top-level `await` available by default.

This MR also already showed that while it's trivial at code level to change these expectations, 5 tests had really hard coded different expectations and I am expecting tests in PyScript to have issue too once this is brought in.

The TL;DR of this MR is that it might take a lot of effort to make this fundamental change around our code but it should be duable, given enough time to test all the things are still working and we don't break all our test coverage we had in place before.

/cc @ntoll

### CODE REVIEW

**Comment by WebReflection** [comment]
*actually* ... I've discovered a bug with `async` scripts and the `document.currentScript` accessor ... so here the thing, we cannot grant that at any point in time the currently executing asynchronous script points at the right `currentScript` within the Python code because async means other scripts might also run in the meantime.

However, I think this is one of those non resolvable cases as these are fighting against the current Web standard implementation of `document.currentScript` and we should never interfere (more than we do already) with that part of the specs.

I think it's safe to move forward with this MR as all tests are green and I am quite satisfied of the result.
